src/content/docs/fr/guides/integrations-guide/cloudflare.mdx

```diff
@@ -0,0 +1,403 @@
+        "/users/faq/",
+        "/favicon.png",
+        "/company/index.html",
+        "/users/faq/index.html"
+      ]
+    }
+    ```
+
+### routes.include
+
+`routes.include: string[]`
+
+Par défaut `[]`
+
+Si vous voulez utiliser la génération automatique de `_routes.json`, mais que vous voulez inclure des routes supplémentaires (par exemple, lorsque vous avez des fonctions personnalisées dans le dossier `functions`), vous pouvez utiliser l'option `routes.include` pour ajouter des routes supplémentaires au tableau `include`.
+
+### routes.exclude
+
+`routes.exclude: string[]`
+
+Par défaut `[]`
+
+Si vous voulez utiliser la génération automatique de `_routes.json`, mais que vous voulez exclure des routes supplémentaires, vous pouvez utiliser l'option `routes.exclude` pour ajouter des routes supplémentaires au tableau `exclude`.
+
+L'exemple suivant génère automatiquement `_routes.json` en incluant et en excluant des routes supplémentaires. Notez que cela n'est nécessaire que si vous avez des fonctions personnalisées dans le dossier `functions` qui ne sont pas gérées par Astro.
+
+```diff lang="ts"
+  // astro.config.mjs
+  export default defineConfig({
+    adapter: cloudflare({
+      mode: 'directory',
++     routes: {
++       strategy: 'include',
++       include: ['/users/*'], // handled by custom function: functions/users/[id].js
++       exclude: ['/users/faq'], // handled by static page: pages/users/faq.astro
++     },
+    }),
+  });
+```
+
+## Activer la prévisualisation
+
+Pour que la prévisualisation fonctionne, vous devez installer `wrangler`
+
+```sh
+pnpm install wrangler --save-dev
+```
+
+Il est alors possible de mettre à jour le script de prévisualisation dans votre `package.json` en `"preview": "wrangler pages dev ./dist"`. Cela vous permettra d'exécuter votre application entière localement avec [Wrangler](https://github.com/cloudflare/wrangler2), qui supporte les secrets, les variables d'environnement, les espaces de noms KV, les objets durables et [tous les autres liens Cloudflare supportés](https://developers.cloudflare.com/pages/platform/functions/#adding-bindings).
+
+## Accès au moteur d'exécution de Cloudflare
+
+Vous pouvez accéder à toutes les liaisons Cloudflare et aux variables d'environnement à partir des composants Astro et des routes API via `Astro.locals`.
+
+Si vous êtes dans un fichier `.astro`, vous accédez au runtime en utilisant le global `Astro.locals` :
+
+```astro
+const env = Astro.locals.runtime.env;
+```
+
+Depuis un point de terminaison (endpoint):
+
+```js
+// src/pages/api/someFile.js
+export function GET(context) {
+  const runtime = context.locals.runtime;
+
+  return new Response('Some body');
+}
+```
+
+En fonction de votre mode d'adaptation (advanced = worker, directory = pages), l'objet runtime sera légèrement différent en raison des différences dans l'API de Cloudflare.
+
+Si vous utilisez le mode d'exécution `advanced`, vous pouvez taper l'objet `runtime` comme suit :
```

Comment:
I don't think we can hurt the `runtime` object. I think we should just define its type :rofl: 
```suggestion
Si vous utilisez le mode d'exécution `advanced`, vous pouvez typer l'objet `runtime` comme ceci :
```